### DISCUSSION TITLE
Traffic dropping on rekey

### DISCUSSION THREAD

**Discussion by doug-tucker-nav** [strongswan/strongswan#1386] [General]
I have a strongswan instance connected to 2 peers at a remote location.  Having an odd issue with on peer where after a rekey, the traffic stops coming from the customer network even though the tunnel is established.  If I initiate traffic from my end, the traffic begins to come in from theirs.  I just noticed this in testing, I have no idea why this seems to open the traffic flow from them.  So as a work around, I have cronned a ping the runs ever few minutes to catch it if a rekey happens as as work around.  I ran a manual rekey request to both connections to see if there was any difference in the logs.  What I see is with the connection that works, there is a delete for the old child immediately following the establishment followed by a closing.  On the one where the traffic stops unless I send a ping, there is no delete following by the closing after establishment.  About 15 minutes later in the logs, there will be a "closing expired" for the old connection.  I'm not clear what is causing this as the configs on the strongswan side are identical for both, except for the peer, subnet, and crypto.  Here are the log entries on the forced rekey of each if this helps.

traffic works:

Nov  1 18:22:06 ip-1.1.1.1 charon: 16[CFG] vici rekey CHILD_SA 'srp-main'
Nov  1 18:22:06 ip-1.1.1.1 charon: 16[ENC] generating QUICK_MODE request 590848858 [ HASH SA No KE ID ID ]
Nov  1 18:22:06 ip-1.1.1.1 charon: 16[NET] sending packet: from 1.1.1.1[4500] to 3.3.3.3[4500] (380 bytes)
Nov  1 18:22:06 ip-1.1.1.1 charon: 13[NET] received packet: from 3.3.3.3[4500] to 1.1.1.1[4500] (364 bytes)
Nov  1 18:22:06 ip-1.1.1.1 charon: 13[ENC] parsed QUICK_MODE response 590848858 [ HASH SA No KE ID ID ]
Nov  1 18:22:06 ip-1.1.1.1 charon: 13[CFG] selected proposal: ESP:AES_CBC_128/HMAC_SHA1_96/MODP_1536/NO_EXT_SEQ
Nov  1 18:22:06 ip-1.1.1.1 charon: 13[IKE] CHILD_SA srp-main{8} established with SPIs c8da6d3b_i 6d390d89_o and TS 10.1.0.0/16 === 10.3.0.0/16
Nov  1 18:22:06 ip-1.1.1.1 charon: 13[ENC] generating QUICK_MODE request 590848858 [ HASH ]
Nov  1 18:22:06 ip-1.1.1.1 charon: 13[NET] sending packet: from 1.1.1.1[4500] to 3.3.3.3[4500] (60 bytes)
Nov  1 18:22:07 ip-1.1.1.1 charon: 14[NET] received packet: from 3.3.3.3[4500] to 1.1.1.1[4500] (76 bytes)
Nov  1 18:22:07 ip-1.1.1.1 charon: 14[ENC] parsed INFORMATIONAL_V1 request 2608762291 [ HASH D ]
Nov  1 18:22:07 ip-1.1.1.1 charon: 14[IKE] received DELETE for ESP CHILD_SA with SPI 06089568
Nov  1 18:22:07 ip-1.1.1.1 charon: 14[IKE] closing CHILD_SA srp-main{5} with SPIs c45dfcb7_i (23166 bytes) 06089568_o (28120 bytes) and TS 10.1.0.0/16 === 10.3.0.0/16

traffic stops: 

Nov  1 17:56:42 ip-1.1.1.1 charon: 11[CFG] vici rekey CHILD_SA 'srp-new'
Nov  1 17:56:42 ip-1.1.1.1 charon: 11[ENC] generating QUICK_MODE request 2387397243 [ HASH SA No KE ID ID ]
Nov  1 17:56:42 ip-1.1.1.1 charon: 11[NET] sending packet: from 1.1.1.1[500] to 2.2.2.2[500] (396 bytes)
Nov  1 17:56:42 ip-1.1.1.1 charon: 05[NET] received packet: from 2.2.2.2[500] to 1.1.1.1[500] (380 bytes)
Nov  1 17:56:42 ip-1.1.1.1 charon: 05[ENC] parsed QUICK_MODE response 2387397243 [ HASH SA No KE ID ID ]
Nov  1 17:56:42 ip-1.1.1.1 charon: 05[CFG] selected proposal: ESP:AES_CBC_256/HMAC_SHA2_256_128/MODP_1536/NO_EXT_SEQ
Nov  1 17:56:42 ip-1.1.1.1 charon: 05[IKE] CHILD_SA srp-new{7} established with SPIs c858397d_i 805bf683_o and TS 10.1.0.0/16 === 10.2.0.0/16
Nov  1 17:56:42 ip-1.1.1.1 charon: 05[ENC] generating QUICK_MODE request 2387397243 [ HASH ]
Nov  1 17:56:42 ip-1.1.1.1 charon: 05[NET] sending packet: from 1.1.1.1[500] to 2.2.2.2[500] (76 bytes)
Nov  1 17:56:46 ip-1.1.1.1 charon: 10[IKE] sending DPD request

**Comment by tobiasbrunner** [comment]
Could just be a problem of the peer implementation with rekeying. In that case disable it or increase the interval. IKEv1 is a crappy protocol, so don't expect it to run smoothly. Of course, the ping is fine too if that works for you.

strongSwan, by default, keeps rekeyed IKEv1 CHILD_SAs installed until they expire, as deletes can get lost (they are unconfirmed, unidirectional messages in IKEv1) and don't even have to be sent in the first place. This can be changed by enabling `charon.delete_rekeyed`, which causes strongSwan to delete the old CHILD_SA immediately after the rekeying. Not sure if that would change anything, though, as you wrote that it's the inbound traffic that doesn't get through.

**Comment by doug-tucker-nav** [comment]
I noticed from the log even their phase2 traffic was using port 500 indicating they did not have nat-t enabled.  I had them enable it, and the issue is resolved.  I have no idea why me initiating a ping would result in opening the traffic flow when they had nat-t off, but all is working as expected now.
